src/fallible_rc.rs

```diff
@@ -0,0 +1,69 @@
+use std::cell::Cell;
+use std::collections::TryReserveError;
+use std::marker::PhantomData;
+use std::ops::Deref;
+use std::ptr::NonNull;
+
+use fallible_collections::FallibleBox;
+
+struct RcInner<T> {
+    rc: Cell<usize>,
+    value: T,
+}
+
+pub struct Rc<T> {
```

Comment:
We should probably explicitly unimplement `Send` and `Sync`.

Reply:
I guess. I thought it would be enough if they were inherited from `T`, but you make a good point.

Reply:
I added this, but it now requires the `optin_builtin_traits` nightly feature.